squire_lib/src/tournament.rs

```diff
@@ -971,8 +980,10 @@ impl TournamentSeed {
     }
 }
 
-impl From<TournamentSeed> for Tournament {
-    fn from(seed: TournamentSeed) -> Self {
+impl TryFrom<TournamentSeed> for Tournament {
```

Comment:
I do not like this approach for a few reasons, but they all stem from the same central problem: you should try to validate user input (of any kind) as soon as possible. This means that the internals of your system don't need to worry about all of the ways that an input can be invalid.

In this case, you can't create a tournament without a seed, so the earliest point to validate input is during the creation of the seed. Putting the check there means that tournament creation will always succeed, which means there is less room for error (and less refactoring) to do in places like the `TournamentManager` which needs to be able to reconstruct a tournament from the ground up.